specs/driver_spec.rb

```diff
@@ -72,9 +72,102 @@
       average.must_be :<=, 5.0
     end
 
-    it "returns zero if no trips" do
+    it "returns nil if no trips" do
       driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV", vin: "1C9EVBRM0YBC564DZ")
-      driver.average_rating.must_equal 0
+      driver.average_rating.must_equal nil
+    end
+  end
+
+  describe 'calculate_total_rev method' do
+    it 'can return total money driver made' do
+      start_time = Time.parse('2015-05-20T12:14:00+00:00')
+      end_time = start_time + 25 * 60 # 25 minutes
+      driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV", vin: "1C9EVBRM0YBC564DZ")
+      trip1 = RideShare::Trip.new({id: 8, driver: driver, start_time: start_time, end_time: end_time, cost: 15.90, passenger: nil, date: "2016-08-08", rating: 5})
+      trip2 = RideShare::Trip.new({id: 8, driver: driver, start_time: start_time, end_time: end_time, cost: 12.10, passenger: nil, date: "2016-08-08", rating: 3})
+      trip3 = RideShare::Trip.new({id: 10, start_time: start_time, end_time: nil, driver: driver, cost: nil, passenger: nil, date: "2016-08-010", rating: nil})
+      driver.add_trip(trip1)
+      driver.add_trip(trip2)
+      driver.add_trip(trip3)
+
+      driver.calculate_total_rev.must_be_kind_of Float
+      driver.calculate_total_rev.must_equal 24.7
+    end
+  end
+
+  describe 'total_drive_time' do
+    it 'can return total time driver drove' do
+      start_time = Time.parse('2015-05-20T12:14:00+00:00')
+      end_time = start_time + 25 * 60 # 25 minutes
+      driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV", vin: "1C9EVBRM0YBC564DZ")
+      trip1 = RideShare::Trip.new({id: 8, start_time: start_time, end_time: end_time, driver: driver, cost: 15.90, passenger: nil, date: "2016-08-08", rating: 5})
+      trip2 = RideShare::Trip.new({id: 9, start_time: start_time, end_time: end_time, driver: driver, cost: 12.10, passenger: nil, date: "2016-08-08", rating: 3})
+      trip3 = RideShare::Trip.new({id: 10, start_time: start_time, end_time: nil, driver: driver, cost: nil, passenger: nil, date: "2016-08-010", rating: nil})
+      driver.add_trip(trip1)
```

Comment:
A lot of this code is duplicated between tests. It might make sense to move it do a `begin` block up in the highest level `describe`, which would make it available in all sub-`describe`s.